pkg/net/net.go

```diff
@@ -98,12 +95,18 @@ func (o *Overlay) gossipLoop() {
 		case <-o.proc.Closing():
 			return
 		case <-ticker.C:
-			p, err := o.n.RandPeer()
+			p, err := o.n.RandPeer() // err is non-nil iff neighborhood is empty
 			if err != nil {
 				continue
 			}
-			recs, err := gossip{
-				o.n, o.h, o.h, p, proto, o.r,
+			s, err := o.h.NewStream(o.ctx(), p, gossipProto)
```

Comment:
This actually needs to be `protoutil.Join(o.proto, "gossip")` because we need the version string and namespace in the protocol path.  I.e., we need something like `/casm/net/0.1.1/mynamespace/gossip`.

`gossipProto` (and `joinProto`) are just used for (un)registering stream handlers, as keys in a dict.  They don't get sent on the wire.